tg/controllers.py

```diff
@@ -70,6 +70,32 @@ def __init__(self, model: Model, view: View, tg: Tdlib) -> None:
         self.tg = tg
         self.chat_size = 0.5
 
+    @bind(msg_handler, ["o"])
+    def open_url(self):
+        msg = self.model.current_msg
+        msg = MsgProxy(msg)
+        if not msg.is_text:
+            self.present_error("Does not contain urls")
+            return
+        text = msg["content"]["text"]["text"]
+        urls = []
+        for entity in msg["content"]["text"]["entities"]:
+            if entity["type"]["@type"] != "textEntityTypeUrl":
+                continue
+            offset = entity["offset"]
+            length = entity["length"]
+            url = text[offset : offset + length]
+            urls.append(url)
+        if not urls:
+            self.present_error("No url to open")
+            return
+        if len(urls) == 1:
+            with suspend(self.view) as s:
+                s.call(f"open -a Firefox '{url}'")
+            return
+        with suspend(self.view) as s:
+            s.run_with_input("urlview", "\n".join(urls))
```

Comment:
it should be added to the Readme as a requirement so use will know that it should install it to make this working